File: ChimpokomonPython/ModuloSRC/Chimpokomons/ChimpokomonFactory.py
from Ataque.CañonConico import CañonConico
from Ataque.GarraMecanica import GarraMecanica
from Ataque.PomadaWassington import PomadaWassington
from Ataque.Zapatazo import Zapatazo
from Ataque.RayoVeloz import RayoVeloz
from Chimpokomons.Chimpokomon import Chimpokomon

class ChimpokomonFactory():

    @staticmethod
    def Carnotron(naturaleza):
        ataques = []
        ataques.append(CañonConico())
        ataques.append(RayoVeloz())
        return Chimpokomon("Carnotron", 20, naturaleza, ataques)

    @staticmethod
    def Gallotronix(naturaleza):
        ataques = []
        ataques.append(GarraMecanica())
        return Chimpokomon("Gallotronix", 25, naturaleza, ataques)

    @staticmethod
    def Zapato(naturaleza):
        print(naturaleza)
        ataques = []
        ataques.append(Zapatazo())
        return Chimpokomon("Zapato", 30, naturaleza, ataques)

    @staticmethod
    def ZapatoConDosAtaques(naturaleza):
        ataques = []
        ataques.append(Zapatazo())
        ataques.append(PomadaWassington())
        return Chimpokomon("ZapatoConDosAtaques", 30, naturaleza, ataques)

File: videojuegoDeChinpokomon/src/chimpokomon/ChimpokomonFactory.java
package Chimpokomon;

import Ataque.*;
import Naturaleza.Naturaleza;

import java.util.ArrayList;
import java.util.List;

public class ChimpokomonFactory {

    public static Chimpokomon Carnotron(Naturaleza Naturaleza) {
        List<Ataque> ataques = new ArrayList<Ataque>(2);
        ataques.add(new CañonConico(4, "CañonConico"));
        ataques.add(new RayoVeloz(3, "RayoVeloz"));
        return new Chimpokomon(ataques, Naturaleza, "Carnotron");
    }

    public static Chimpokomon Gallotronix(Naturaleza Naturaleza) {
        List<Ataque> ataques = new ArrayList<Ataque>(2);
        ataques.add(new GarraMecanica(2, "GarraMecanica"));
        return new Chimpokomon(ataques, Naturaleza, "Gallotronix");
    }

    public static Chimpokomon Zapato(Naturaleza Naturaleza) {
        List<Ataque> ataques = new ArrayList<Ataque>(1);
        ataques.add(new Zapatazo(1, "Zapatazo"));
        return new Chimpokomon(ataques, Naturaleza, "Zapato");
    }

    public static Chimpokomon ZapatoConDosAtaques(Naturaleza Naturaleza) {
        List<Ataque> ataques = new ArrayList<Ataque>(2);
        ataques.add(new Zapatazo(1, "Zapatazo"));
        ataques.add(new PomadaWassington(0, "PomadaWassington"));
        return new Chimpokomon(ataques, Naturaleza, "ZapatoConDosAtaques");
    }

}
